libmcount/dynamic.c

```diff
@@ -643,6 +643,74 @@ bool mcount_add_badsym(struct mcount_dynamic_info *mdi, unsigned long callsite,
 	return true;
 }
 
+void mcount_handle_dlopen(struct symtabs *symtabs, struct dl_phdr_info *info,
+			  char *mod_realpath, size_t size)
+{
+	struct uftrace_mmap *map;
+	struct mcount_dynamic_info *mdi;
+	char *libname, *modname;
+	unsigned long vaddr = 0, text_addr = 0;
+	unsigned int text_size = 0, namelen, i;
+	bool text_section_found = false;
+
+	modname = basename(mod_realpath);
+
+	if (!find_module_list(modname))
+		return;
+
+	libname = xstrdup(modname);
+	namelen = ALIGN(strlen(libname) + 1, 4);
+	map = xzalloc(sizeof(*map) + namelen);
+	map->start = info->dlpi_addr;
+	map->len = namelen;
+	mcount_memcpy1(map->libname, libname, namelen);
+	map->libname[namelen] = '\0';
+
+	map->end = 0;
```

Comment:
This is not needed since map is allocated by xzalloc() which sets all memory to zero (that's why it has the 'z' prefix).

Reply:
I did not know it! it has be changed.

Reply:
??

Reply:
I didn't know your saying has include code that giving 0 to the map->end. it changed.